native/acter/src/api/attachments.rs

```diff
@@ -238,123 +238,233 @@ impl AttachmentsManager {
     }
 
     #[allow(clippy::too_many_arguments)]
-    pub fn image_attachment_draft(
+    pub async fn image_attachment_draft(
         &self,
         body: String,
-        url: String,
-        mimetype: Option<String>,
+        url_or_path: String,
         size: Option<u64>,
         width: Option<u64>,
         height: Option<u64>,
         blurhash: Option<String>,
     ) -> Result<AttachmentDraft> {
-        let Room::Joined(joined) = &self.room else {
+        let client = self.client.clone();
+        let Room::Joined(room) = self.room.clone() else {
             bail!("Can only attachment in joined rooms");
         };
-        let info = assign!(ImageInfo::new(), {
-            height: height.and_then(UInt::new),
-            width: width.and_then(UInt::new),
-            mimetype,
-            size: size.and_then(UInt::new),
-            blurhash,
-        });
-        let url = Box::<MxcUri>::from(url.as_str());
-        let mut builder = self.inner.draft_builder();
+        let r = room.clone();
+        let image_content = RUNTIME
+            .spawn(async move {
+                let url = Box::<MxcUri>::from(url_or_path.as_str()); // http not allowed for remote url
+                if url.is_valid() {
+                    return anyhow::Ok(ImageMessageEventContent::plain(body, url.into()));
+                }
+                let path = PathBuf::from(url_or_path.clone());
+                let guess = mime_guess::from_path(path.clone());
+                let content_type = guess.first().context("No MIME type")?;
```

Comment:
same as above, let's give the user a bit more information:
```suggestion
                let content_type = guess.first().context("Detecting mime type failed. Are you sure this is an Image?")?;
```

Applies to the other functions that follows as well, of course.